tests/test_arrow_dataset.py

```diff
@@ -750,9 +750,11 @@ def test_flatten(self, in_memory):
             ) as dset:
                 with self._to(in_memory, tmp_dir, dset) as dset:
                     fingerprint = dset._fingerprint
+                    dset._format_columns = ["a", "foo"]
```

Comment:
Could you add a test where "a" is not formatted, and make sure "a.b.c" stays unformatted as well ?

I think right now this would fail, so we need to use the same trick as in `rename_columns`

https://github.com/huggingface/datasets/blob/ede72d3f9796339701ec59899c7c31d2427046fb/src/datasets/arrow_dataset.py#L1759-L1760

Reply:
I think that if you set `dset._format_columns = ["foo"]` and then call `flatten`, then the new column `a.b.c` will end up being formatted (i.e. inside `_format_columns`). However I would expect `a.b.c` to be unformatted, since `a` was not part of the formatted columns before flatten.

We had a similar issue with `rename`: when an unformatted column "a" was renamed to "b", then "b" would have been formatted. To fix this issue we explicitly set `dataset._format_columns` inside `rename`, even though it already has `@transmit_columns`

I hope it's clearer this way, let me know if you have questions or if I can help